src/routing_node.rs

```diff
@@ -137,7 +138,7 @@ impl<F> RoutingNode<F> where F: Interface {
     }
 
     /// Add something to the network, will always go via ClientManager group
-    pub fn put<T>(&mut self, destination: NameType, content: T) where T: Sendable {
+    pub fn put(&mut self, destination: NameType, content: Box<Sendable>, is_client: bool) {
```

Comment:
can you rename is_client to client_authority plz.
Also you are missing this from my code.

        let authority = match client_authority {
            true => types::Authority::Client,
            false => types::Authority::ManagedNode,
        };